Nos/Views/CompactNoteView.swift

```diff
@@ -97,11 +102,13 @@ struct CompactNoteView: View {
                     .onPreferenceChange(TruncatedSizePreferenceKey.self) { newSize in
                         if newSize.height > truncatedSize.height {
                             truncatedSize = newSize
-                            updateShouldShowReadMore()
                         }
                     }
                     .background {
-                        noteText
+                        // To calculate whether the text should be truncated we create this hidden view of 
+                        // `formattedText` and record its size. It is compared to `truncatedSize` and used in the 
+                        // calculation of `noteNeedsTruncation`.
+                        formattedText
```

Comment:
I think this is the part that actually fixed the bug. The rest of the diff is refactoring to make things clearer.